components/series/stackedBar.js

```diff
@@ -6,77 +6,39 @@
-            // converts it into an array of values. Each value has the xValue associated
-            // with it.
-            //
-            // For example, this object:
-            //
-            // obj = { county: 'North Tyneside', labour: 23, conservative: 55 }
-            //
-            // becomes this ...
-            //
-            // [
-            //   { xValue: 'North Tyneside', name: 'labour', previousValue: 0, currentValue: 23},
-            //   { xValue: 'North Tyneside', name: 'conservative', previousValue: 23, currentValue: 78},
-            // ]
-            function objectDatapointToArray(obj) {
-                var values = [];
-                var yTotal = 0;
-                var xVal = obj[stackedBar.xValueKey.value];
-                for (var propertyName in obj) {
-                    if (obj.hasOwnProperty(propertyName) && propertyName !== stackedBar.xValueKey.value) {
-                        var previous = yTotal;
-                        yTotal += Number(obj[propertyName]);
-                        values.push({
-                            'name': propertyName,
-                            'previousValue': previous,
-                            'currentValue': yTotal,
-                            'xValue': xVal
-                        });
-                    }
-                }
-                return values;
-            }
-
             selection.each(function(data) {
 
                 // add a 'root' g element on the first enter selection. This ensures
                 // that it is just added once
                 container = d3.select(this);
 
-                var keyFunction = function(d) {
-                    return d[stackedBar.xValueKey.value];
-                };
-                var g = fc.utilities.simpleDataJoin(container, 'stacked-bar', data, keyFunction);
+                // If the user has already set stacked data then we don't need to re-compute it.
+                var layers = stackedBar.stackedData.value;
+                if (typeof layers === 'undefined' || layers.length === 0) {
+                    layers = stackedBar.layout.value(data);
+                }
+
+                var g = fc.utilities.simpleDataJoin(container, 'stacked-bar', layers);
 
+                stackedBar.decorate.value(g);
 
-                // create a join for each bar
                 var bar = g.selectAll('rect')
-                    .data(function(d) { return objectDatapointToArray(d); })
+                    .data(function(d) { return d; })
                     .enter()
                     .append('rect');
 
-                // compute the bar width from the x values
-                var xValues = data.map(function(d) {
-                    return stackedBar.xScale.value(d[stackedBar.xValueKey.value]);
-                });
+                // Compute the bar width from the x values
+                // Assumes first series contains all possible X values.
```

Comment:
No reason other than simplicity. I could create a set of x values from all of the provided arrays. In hindsight, this probably wouldn't be much more difficult. I will update!